pierg/fact-fortress-dapp
# [WIP] Support Schnorr signatures verification

**[WIP] Support Schnorr signatures verification**

Attempting to verify onchain Schnorr signatures

- Generate proof via CLI and programmatically ✅ 
- Verify proof via CLI and programmatically ✅ 
- Verify proof via smart contract generated by CLI 🟥 (see: [https://github.com/noir-lang/noir/issues/1133](https://github.com/noir-lang/noir/issues/1133))
- Verify proof via smart contract generated programmatically and tweaked manually ✅


To test:
```
git checkout schnorr_signature
cd smart_contracts/Ethereum/ZkpContract
pnpm i
pnpm run test
```
